nixos/modules/services/x11/redshift.nix

```diff
@@ -88,51 +71,101 @@ in {
       example = "/bin/redshift-gtk";
       description = ''
         Redshift executable to use within the package.
+
+        To use Gammastep, pass either <literal>/bin/gammastep</literal>
+        or <literal>/bin/gammastep-indicator</literal>.
       '';
     };
 
-    extraOptions = mkOption {
-      type = types.listOf types.str;
-      default = [];
-      example = [ "-v" "-m randr" ];
+    settings = mkOption {
+      type = types.submodule {
+        freeformType = settingsFormat.type;
+      };
+
+      default = {};
+
+      example = {
+        redshift = {
+          temp-day = 5500;
+          temp-night = 3700;
+          brightness-day = 1.0;
+          brightness-night = 0.8;
+          fade = 1;
+        };
+        randr = {
+          screen = 0;
+        };
+      };
+
       description = ''
-        Additional command-line arguments to pass to
-        <command>redshift</command>.
+        The configuration to pass to Redshift/Gammastep.
+
+        Available options for Redshift described in
+        <citerefentry>
+          <refentrytitle>redshift</refentrytitle>
+          <manvolnum>1</manvolnum>
+        </citerefentry> and for Gammastep in
+        <citerefentry>
+          <refentrytitle>gammastep</refentrytitle>
+          <manvolnum>1</manvolnum>
+        </citerefentry>.
       '';
     };
   };
 
-  config = mkIf cfg.enable {
+  config = let
+    mainSection = if strings.hasInfix "gammastep" cfg.executable
+      # https://gitlab.com/chinstrap/gammastep/-/commit/1608ed61154cc652b087e85c4ce6125643e76e2f
+      then "general"
+      else "redshift";
+    target = if (cfg.settings.${mainSection}.adjustment-method or null) == "drm"
```

Comment:
With above type change, this should check the value for `cfg.settings.general.adjustment-method` (since this will be set whether the user sets `general` or `redshift`.